packages/medusa/src/services/product.ts

```diff
@@ -433,6 +439,21 @@ class ProductService extends TransactionBaseService {
         }
       }
 
+      if (isDefined(categories)) {
+        product.categories = []
+
+        if (categories?.length) {
+          const categoryIds = categories?.map((c) => c.id)
+          const pcRepo = manager.getCustomRepository(this.productCategoryRepository_)
+          const categoryRecords = await pcRepo.find({
```

Comment:
todo: we should use the productCategoryService here. It means adding the list method to the service since we don't want the count. Also, if some of the categories that are provided do not exists, the user is not informed. Should we handle that?

cc: @olivermrbl

Reply:
Could we get rid of the repository used by replacing this with:
```
const categoryRecords = categoryIds.map((c) => ({ id: c } as ProductCategory))
```

And then let it throw on FK constraint violation?

Reply:
Yes we can indeed 💪